tests/test_rotate_visualisations.py

```diff
@@ -0,0 +1,128 @@
+from tests.helper import requires
+from swiftsimio.subset_writer import find_links, write_metadata
+from swiftsimio import mask, cosmo_array, load
+from swiftsimio.visualisation.projection import project_gas
+from swiftsimio.visualisation.slice import slice_gas
+from swiftsimio.visualisation.volume_render import render_gas
+from swiftsimio.visualisation.rotation import rotation_matrix_from_vector
+import h5py
+from numpy import array_equal, mean
+
+@requires("cosmological_volume.hdf5")
+def test_rotation(filename):
+    output = "single_particle.hdf5"
+    
+    data_mask = mask(filename)
+    boxsize = data_mask.metadata.boxsize
+    region = [[0, b] for b in boxsize]
+    data_mask.constrain_spatial(region)
+    
+    # Write the metadata
+    infile = h5py.File(filename, "r")
+    outfile = h5py.File(output, "w")
+    list_of_links, _ = find_links(infile)
+    write_metadata(infile, outfile, list_of_links, data_mask)
+    
+    # Write a single particle
+    particle_coords = cosmo_array([[1,1,1], [1,1,1]], data_mask.metadata.units.length)
+    particle_masses = cosmo_array([1, 1], data_mask.metadata.units.mass)
+    mean_h = mean(infile["/PartType0/SmoothingLengths"])
+    particle_h = cosmo_array([mean_h, mean_h], data_mask.metadata.units.length)
+    particle_ids = [1, 2]
+    
+    coords = outfile.create_dataset("/PartType0/Coordinates", data=particle_coords)
+    for name, value in infile["/PartType0/Coordinates"].attrs.items():
+        coords.attrs.create(name, value)
+    
+    masses = outfile.create_dataset("/PartType0/Masses", data=particle_masses)
+    for name, value in infile["/PartType0/Masses"].attrs.items():
+        masses.attrs.create(name, value)
+    
+    h = outfile.create_dataset("/PartType0/SmoothingLengths", data=particle_h)
+    for name, value in infile["/PartType0/SmoothingLengths"].attrs.items():
+        h.attrs.create(name, value)
+    
+    ids = outfile.create_dataset("/PartType0/ParticleIDs", data=particle_ids)
+    for name, value in infile["/PartType0/ParticleIDs"].attrs.items():
+        ids.attrs.create(name, value)
+    
+    outfile.create_dataset("/PartType1/Coordinates", shape = particle_coords.shape)
+    
+    # Tidy up
+    infile.close()
+    outfile.close()
+    
+    # Start from the beginning, open the file 
+    data = load(output)
```

Comment:
I suppose it's easier here to use this file multiple times, but it would be nice to have the differetn visualisation funtionality tested in different functions.

Can you add the code that creates a single particle file to the helper, make it in-memory, and then use this all to test in individual functions?